File: VB/PropertyGrid_Sorting/Person.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq

Namespace PropertyGrid_Sorting
	Public Class Person
		<Category("Info")>
		Public Property ID() As Integer
		<Category("Info")>
		Public Property FirstName() As String
		<Category("Info")>
		Public Property LastName() As String
		<Category("Contact"), TypeConverter(GetType(ExpandableObjectConverter))>
		Public Property Address() As SimpleAddress
		<Category("Contact")>
		Public Property Email() As String
		<Category("Info")>
		Public Property Orders() As List(Of Order)
	End Class

	Public Class SimpleAddress
		Public Property State() As String
		Public Property City() As String
		Public Property ZIP() As String
	End Class

	Public Class Order
		Public Property ID() As Integer
		Public Property ContactName() As String
		Public Property OrderDate() As Date
	End Class
End Namespace

File: CS/PropertyGrid_Sorting/Person.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;

namespace PropertyGrid_Sorting
{
    public class Person
    {
        [Category("Info")]
        public int ID { get; set; }
        [Category("Info")]
        public string FirstName { get; set; }
        [Category("Info")]
        public string LastName { get; set; }
        [Category("Contact")]
        [TypeConverter(typeof(ExpandableObjectConverter))]
        public SimpleAddress Address { get; set; }
        [Category("Contact")]
        public string Email { get; set; }
        [Category("Info")]
        public List<Order> Orders { get; set; }
    }

    public class SimpleAddress
    {
        public string State { get; set; }
        public string City { get; set; }
        public string ZIP { get; set; }
    }

    public class Order
    {
        public int ID { get; set; }
        public string ContactName{ get; set; }
        public DateTime OrderDate { get; set; }
    }
}
